Source/WebCore/platform/graphics/ImageBuffer.h

```diff
@@ -49,11 +49,11 @@ enum class ImageBufferOptions : uint8_t {
     UseDisplayList  = 1 << 1
 };
 
-class ImageBuffer : public ThreadSafeRefCounted<ImageBuffer, WTF::DestructionThread::Main>, public CanMakeWeakPtr<ImageBuffer> {
```

Comment:
The changes for ImageBuffer to use DestructionThread::Main were to handle RemoteRenderingBackendProxy (in the WebContent process). Previously we allowed these to be sent to a worker, and if they got destroyed there they attempted to use IPC from the wrong thread.

This patches solves this by making sure we don't copy the RRBP between thread, and remote proxy buffers now strongly belong to their thread, so DestructionThread::Main is incorrect.

In the GPUP, DestructionThread::Main wasn't necessary (as far as I can tell), and potentially causes issues with WeakPtrs (which aren't threadsafe), since we hold weak refs and send them to be destroyed on a different thread.